File: code-examples/examples/example-2/src/main/java/be/swsb/typesafety/example2/service/booking/BookingService.java
package be.swsb.typesafety.example2.service.booking;

import be.swsb.typesafety.example2.domain.booking.Booking;
import be.swsb.typesafety.example2.domain.booking.BookingRepository;

import java.util.UUID;

import static be.swsb.typesafety.example2.domain.booking.Booking.booking;

public class BookingService {

    private final BookingRepository repo;

    public BookingService(final BookingRepository repo) {
        this.repo = repo;
    }

    public UUID createNewBooking(final String bookingCode) {
        Booking booking = booking(bookingCode);
        return repo.save(booking);
    }
}

File: code-examples/examples/example-5/src/main/java/be/swsb/typesafety/example5/service/booking/BookingService.kt
package be.swsb.typesafety.example5.service.booking

import be.swsb.typesafety.example5.domain.booking.Booking.Companion.booking
import be.swsb.typesafety.example5.domain.booking.BookingCode
import be.swsb.typesafety.example5.domain.booking.BookingRepository
import java.util.*

open class BookingService(private val repo: BookingRepository) {

    open fun createNewBooking(bookingCode: BookingCode?): UUID {
        val booking = booking(bookingCode)
        return repo.save(booking)
    }
}
